crypto/merkle/simple_proof.go

```diff
@@ -49,11 +63,32 @@ func SimpleProofsFromMap(m map[string]Hasher) (rootHash []byte, proofs map[strin
 	return
 }
 
-// Verify that leafHash is a leaf hash of the simple-merkle-tree
-// which hashes to rootHash.
-func (sp *SimpleProof) Verify(index int, total int, leafHash []byte, rootHash []byte) bool {
-	computedHash := computeHashFromAunts(index, total, leafHash, sp.Aunts)
-	return computedHash != nil && bytes.Equal(computedHash, rootHash)
+// Verify that the SimpleProof proves the root hash.
+func (sp *SimpleProof) Verify(rootHash []byte, index int, total int, leafHash []byte) error {
```

Comment:
Can we exclude `index int, total int, leafHash []byte` from the argument list? The fields are exported, the user can check it manually when they need to. Or we can make a new method like `Validate` to check the fields' validity.

Reply:
I think we can exclude index/total, but we should leave rootHash and leafHash.  The other two we can expose GetIndex() and GetTotal().  We should add a comment that the user should check these if they need to verify this info.